packages/shared/lib/services/connections/credentials/refresh.ts

```diff
@@ -111,9 +113,24 @@ export async function refreshOrTestCredentials(props: RefreshProps): Promise<Res
 
         if (res.isErr()) {
             span.setTag('error', res.error);
+            await connectionService.setRefreshFailure({ id: props.connection.id, currentAttempt: props.connection.refresh_attempts || 0 });
             return res;
         }
-        return Ok(res.value);
+
+        let newConnection = res.value;
+        // Backfill
```

Comment:
not sure I understand the backfill comment

Reply:
true, I'll add a better comment